visidata/data.py

```diff
@@ -183,31 +186,40 @@ def loadInternalSheet(klass, p, **kwargs):
     return vs
 
 @Sheet.api
-def deleteBy(self, func):
+def deleteBy(self, func, commit=False):
```

Comment:
This `commit` variable got added because `addNewRows` (see above) uses `deleteBy` in its undo. It essentially forces a committed delete, even if defer=True.

We want that if a user undoes a deferredAdd, that that added row just disappears. We do not want it turning red. 

It could possibly be handled better than this. Or the variable could be better named.